lock/src/main/java/org/infinispan/lock/impl/ClusteredLockModuleLifecycle.java

```diff
@@ -0,0 +1,100 @@
+import org.infinispan.lock.impl.functions.LockFunction;
+import org.infinispan.lock.impl.functions.UnlockFunction;
+import org.infinispan.lock.impl.lock.ClusteredLockFilter;
+import org.infinispan.lock.impl.manager.CacheHolder;
+import org.infinispan.lock.impl.manager.EmbeddedClusteredLockManager;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.partitionhandling.PartitionHandling;
+import org.infinispan.registry.InternalCacheRegistry;
+import org.infinispan.transaction.TransactionMode;
+import org.kohsuke.MetaInfServices;
+
+/**
+ * Locks module configuration
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ * @since 9.2
+ */
+@MetaInfServices(value = ModuleLifecycle.class)
+public class ClusteredLockModuleLifecycle implements ModuleLifecycle {
+
+   public static final String CLUSTERED_LOCK_CACHE_NAME = "___clustered_locks";
+
+   @Override
+   public void cacheManagerStarting(GlobalComponentRegistry gcr, GlobalConfiguration globalConfiguration) {
+      final Map<Integer, AdvancedExternalizer<?>> externalizerMap = globalConfiguration.serialization()
+            .advancedExternalizers();
+      externalizerMap.put(ClusteredLockKey.EXTERNALIZER.getId(), ClusteredLockKey.EXTERNALIZER);
+      externalizerMap.put(ClusteredLockValue.EXTERNALIZER.getId(), ClusteredLockValue.EXTERNALIZER);
+      externalizerMap.put(LockFunction.EXTERNALIZER.getId(), LockFunction.EXTERNALIZER);
+      externalizerMap.put(UnlockFunction.EXTERNALIZER.getId(), UnlockFunction.EXTERNALIZER);
+      externalizerMap.put(IsLocked.EXTERNALIZER.getId(), IsLocked.EXTERNALIZER);
+      externalizerMap.put(ClusteredLockFilter.EXTERNALIZER.getId(), ClusteredLockFilter.EXTERNALIZER);
+   }
+
+   @Override
+   public void cacheManagerStarted(GlobalComponentRegistry gcr) {
+      final EmbeddedCacheManager cacheManager = gcr.getComponent(EmbeddedCacheManager.class);
+      final InternalCacheRegistry internalCacheRegistry = gcr.getComponent(InternalCacheRegistry.class);
+
+      internalCacheRegistry.registerInternalCache(CLUSTERED_LOCK_CACHE_NAME, createClusteredLockCacheConfiguration(),
+            EnumSet.of(InternalCacheRegistry.Flag.EXCLUSIVE));
+
+      CompletableFuture<CacheHolder> future = startCaches(cacheManager);
+      registerClusteredLockManager(gcr, future);
+   }
+
+   private static Configuration createClusteredLockCacheConfiguration() {
+      ConfigurationBuilder builder = new ConfigurationBuilder();
+      builder.clustering().cacheMode(CacheMode.REPL_SYNC)
+            .stateTransfer().fetchInMemoryState(true)
+            .partitionHandling().whenSplit(PartitionHandling.DENY_READ_WRITES)
+            .transaction().transactionMode(TransactionMode.NON_TRANSACTIONAL);
+
+      return builder.build();
+   }
+
+   private static CompletableFuture<CacheHolder> startCaches(EmbeddedCacheManager cacheManager) {
+      final CompletableFuture<CacheHolder> future = new CompletableFuture<>();
+      new Thread(() -> {
```

Comment:
Please set a thread name that includes the node name for easier debugging.

Reply:
Ok